### ISSUE TITLE
`docker-compose create` fails with 'ERROR: network xxx_default: not found'

### ISSUE THREAD

**Issue by nschoe** [docker/compose#3477]
When we create and start services containers with either `docker-compose up` or `docker-compose start` it creates a custom network, whose name is `xxx_default` with `xxx` being the current directory or the value of `-p | --project-name`.

But apparently, it doesn't do that for `docker-compose create`.

Have I missed a step, is this deliberate or is it a bug?

**Comment by shin-** [comment]
Related: #3395

**Comment by shin-** [comment]
FWIW,
```
$ docker-compose create --help
Creates containers for a service.
```

The command specifically says it only creates containers. I don't know if this should be considered as a bug.
Any specific reason you're using `create` + `start` instead of `up`?

**Comment by nschoe** [comment]
Yes, sorry I should have mentioned it clearer. I know that the doc says it creates containers. So technically, this is not a bug, but: does it even make sense?

In my case, there _is_ a specific reason. Actually, I use docker to containerize my clients environments. Since my service is comprised of several containers, I use docker-compose for that, because it handily creates a custom network for the services, and make each containers part of this network. This is extremely useful to me.

My registration process is a two-step where the client registers himself, then through activations / tokens, the services are made available for him.
I'd like to be able to _create_ the services, but not start them as long as the client did not use his token.

This is not critical of course, just a matter of convenience :-)

In my sense, `docker-compose create` as it is not very helpful. But then again, it was just a suggestion :-)

**Comment by johnharris85** [comment]
I think the spirit of `create` should be to create everything for the 'context'. So `create` with docker creates the container. `create` for compose should create everything that compose would create if run. 

I guess it's really a philosophical question of whether `create` should match the docker behaviour (containers only) or should match the 'contextual' behavior (engine vs compose).

The reasoning for the latter is that `docker create` + `docker start` = `docker run`, so maybe `docker-compose create` + `docker-compose start` should = `docker-compose up`?

I don't think it should be considered a bug, but a feature request / improvement. Opinions?

**Comment by nschoe** [comment]
I agree with @johnharris85 , sort of. I know this is technically not a bug, but since it's named "create" as in "docker create" and as it's just been said above, `docker run = docker create + docker start`, I think that people will tend to think docker-compose runs the same.

But I'd be perfectly okay to change this into "feature_request" :-)

**Comment by aaronjensen** [comment]
Is there a docker-compose command to create the networks? Or should one manually create them with `docker network create`?

I just ran into this as well.

**Comment by outcoldman** [comment]
I would say that it is technically a bug. 

Docker itself allows you to "create" everything first (with combination of `docker network create` and `docker create`) and only after that start containers. Docker compose does not have this option. Yes we can use `docker network create`, but we do not know which `network` docker-compose will generate. Yes, we might guess that based on folder name and the name specified, but this is not documented anywhere, so technically we cannot rely on this solution as it can be changed in future.

@shin- the reason why you want to do that - after `create` you might want to upload special file to the container itself, like product license or some secure token, which you don't want to inject in the image.

**Comment by aanand** [comment]
This is a duplicate of https://github.com/docker/compose/issues/2908, so I'm going to close it. Discussion on the right approach is already happening both there and in https://github.com/docker/compose/issues/3395, so the fewer places we're having the discussion, the better.

I still think the right solution here is a `--no-start` flag to `up`.
